src/wilcoxon.jl

```diff
@@ -163,6 +163,27 @@ function StatsAPI.pvalue(x::ExactSignedRankTest; tail=:both)
     end
 end
 
+function psignrank(W, n, lower_tail)
+    int_type = typeof(n)
+    W = int_type(W)
+    two = one(int_type) + one(int_type)
+    max_W =  n * (n + one(int_type)) ÷ two + one(int_type)
+    DP = zeros(int_type(n), max_W)
+    DP[begin] = one(int_type)
+    for j in one(int_type):n
+        for i in max_W:-one(int_type):one(int_type)
+            if i - j >= one(int_type)
+                DP[i] += DP[i - j]
+            end
+        end
```

Comment:
This could be simplified to

```suggestion
        for i in max_W:-1:(j + 1)
            DP[i] += DP[i - j]
        end
```